src/auth/dto/login.dto.ts

```diff
@@ -0,0 +1,9 @@
+import { IsNotEmpty } from '@nestjs/class-validator';
+
+export class LoginDto {
+  @IsNotEmpty()
+  login: string;
+
+  @IsNotEmpty()
+  password: string;
```

Comment:
if you using class-validator, why not validate this DTO data at `passing` stage like you did it in user/group create/update DTO? This way you can find wrong data almost immediately even without sending any queries to a database.

Reply:
I don't grasp your idea. The dto is validated when the request is recieved. It is the earliest stage where we can do anything, I assume. No queries will be made if dto is wrong.